src/axolotl/utils/callbacks.py

```diff
@@ -317,3 +320,250 @@ def on_evaluate(
                 trainer.log(results)
 
     return BenchEvalCallback
+
+
+def log_prediction_callback_factory(trainer: Trainer, tokenizer):
+    LOG.info("log_prediction_callback_factory")
+
+    class LogPredictionCallback(TrainerCallback):
+        """Callback to log prediction values during each evaluation"""
+
+        def __init__(self, cfg):
+            self.cfg = cfg
+            self.logged = False
+
+        def on_evaluate(
+            self,
+            args: AxolotlTrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model,
+            # tokenizer,
+            train_dataloader,
+            eval_dataloader,
+            **kwargs,
+        ):
+            LOG.info("=" * 80)
+            LOG.info("logging predictions")
+
+            trainer.model.eval()
+
+            def logits_to_tokens(logits) -> str:
```

Comment:
**Help Wanted**

I feel like I must be reinventing the wheel here. I'd be surprised if there was not a function which did this / similar / or more already, although I do not know what it is. Recommendations welcome!

Reply:
I don't know off the top of my head, would probably need to dig through the transformers code to see what's available